File: DevEnvironment/Command/src/main/java/command/core/InputParameters.java
package command.core;

import base.core.Conditions;
import command.interfaces.IInputParameters;

import java.util.List;
import java.util.Map;

/**
 * The InputParameters class implements input parameters of a command.
 */
public final class InputParameters implements IInputParameters {
    private final List<String> indexedParameters;
    private final Map<String, String> namedParameters;

    /**
     * The InputParameters constructor.
     */
    public InputParameters(
        List<String> indexedParameters,
        Map<String, String> namedParameters) {

        Conditions.validateNotNull(
            indexedParameters,
            "The indexed parameters.");

        Conditions.validateNotNull(
            namedParameters,
            "The named parameters.");

        this.indexedParameters = indexedParameters;
        this.namedParameters = namedParameters;
    }

    /**
     * Gets indexed parameters.
     */
    @Override
    public List<String> getIndexedParameters() {
        return this.indexedParameters;
    }

    /**
     * Gets named parameters.
     */
    @Override
    public Map<String, String> getNamedParameters() {
        return this.namedParameters;
    }
}

File: DevEnvironmentCLion/Command/core/InputParameters.cpp
#include "PreCompiled.h"
#include "InputParameters.h"

using namespace command;

/**
 * Makes new input parameters.
 */
IInputParametersPtr InputParameters::make(
    const std::vector<std::string>& indexedParameters,
    const std::map<std::string, std::string>& namedParameters)
{
    return std::make_shared<InputParameters>(indexedParameters, namedParameters);
}

/**
 * The InputParameters constructor.
 */
InputParameters::InputParameters(
    const std::vector<std::string>& indexedParameters,
    const std::map<std::string, std::string>& namedParameters) :
    m_indexedParameters(indexedParameters),
    m_namedParameters(namedParameters)
{
}

/**
 * The InputParameters destructor.
 */
InputParameters::~InputParameters()
{
}

/**
 * Gets indexed parameters.
 */
const InputParameters::IndexedParameters& InputParameters::getIndexedParameters() const
{
    return m_indexedParameters;
}

/**
 * Gets named parameters.
 */
const InputParameters::NamedParameters& InputParameters::getNamedParameters() const
{
    return m_namedParameters;
}
